### DISCUSSION TITLE
why isn't linux supported on the surface laptop??

### DISCUSSION THREAD

**Discussion by IsAmogusDead** [linux-surface/linux-surface#567] [General]
i mean ... it's just a laptop? Not much different from a thinkpad (at least i think) was it specialy made not to support linux?

**Comment by qzed** [comment]
Well, what's "just a laptop"? Mostly a bunch of hardware stuck together, so it's all about drivers for those components, really. And at this point, the Surface devices are mostly supported on newer kernels with exception of some WiFi fixes (getting there), some other small fixes (mostly either hacks for quirky old devices, unfinished stuff, or new AMD fixes that haven't made it usptream yet), and touchscreen.

I'd argue that, with the exception of touchscreen data processing, any missing upstream support is actually not really a problem with Microsoft (at least not directly). So, no, I don't think MS goes out of their way to especially make the devices unfriendly to Linux (at least not right now...). For example, the missing touch hardware (not processing) support on the SP7+ is really Intel's fault since it's their PCI device that has no driver in the kernel. The WiFi fixes on older devices are an issue with Marvell (now NXP) not taking care of their stuff and having just no interest at all in properly maintaining a kernel driver (and firmware for Linux). The problems with the new AMD versions aren't exclusive to the Surface devices as well, and it seems like AMD is only just now (as in latest 2 major kernel releases or so) working on fixing those. Let's throw in the Surface Pro X for good measure because that's missing a ton of stuff for the Qualcomm SoC. Microsoft doesn't really do much hardware, and I believe all of what MS did contribute hardware-wise (only EC as far as I can tell) is now supported (v5.13+, that is).

Ultimately, I believe the support issue is a combination of two things:

1. MS doesn't care about any sort of validation for Linux. I.e. that the things they choose for their devices will work on other OSes too. That's a thing that Lenovo does for their Thinkpad lineup (I believe). They also do have active kernel developers working on supporting their devices. Other vendors might do something similar, at least for some (probably business) devices (IIRC Dell has also active kernel developers).
2. MS doesn't hesitate to go with somewhat non-standard hardware (which might be related to the first point). The two most prominent examples are their own embedded controller (and corresponding protocol) and the touchscreen via IPTS. I can only speculate as to why they chose to use those:

   At this point I think pretty much all notebooks have an embedded controller (even if just for managing battery health), if not more. In short it's responsible for anything that you may not want the OS to deal with because it's e.g. too critical. There are probably many reasons why it made sense for them to ship their own EC, including maintainability, improved user quality, cost, ... They're a big company and can spend the money up-front for developing something like this if it means they can fix problems later on in software and just re-use the thing or slightly update it from generation to generation instead of having to go with a completely new solution. And they're not the only company doing that. Apple also has their SMC and T2 chip... so there are definitely reasons beyond "let's make something that doesn't work on Linux".

   Touch is a pretty similar situation, and IPTS is actually Intel's brain-child, although it didn't get much use besides the Surface lineup. I think the main reasons there are really independence from OEMs, improved user experience, and more independence from individual device hardware. IPTS does a lot of the processing in software, that is fully configurable by the vendor (not the OEM). So it doesn't rely on some hardware controller doing all the processing where you'd then have to ask the OEM to fix something and provide a new firmware for (if that's even fixable in firmware and it's not hard-wired). Lately, MS is even going a step beyond IPTS with their `TouchPenProcessor.dll` that's responsible for all the processing on newer devices. A big part here is that things get less and less device specific, since the device just sends raw data and the DLL does all the hard processing. That means that you can just re-use the DLL for new devices and don't need a new controller, and also that you can update the DLL for all devices, new and old, and don't need to maintain different software/firmware for each device.

So, again, AFAICT it's not that MS wants to cripple the Linux experience (or goes out of their way to do that), but just that they choose hardware in a way that helps them optimize their internal processes/profit down the line or they find best for the device that they want to create (wrt. user experience, ...). They don't care about Linux, so any choice they make with regards to hardware just isn't influenced by that.

TLDR: The only thing you can blame MS for is that they don't care about Linux (support on their devices, they do care if it makes them money in their server business). I don't think there's any outright malicious intent to that.

**Reply by IsAmogusDead** [reply]
the great Chinese wall of text

**Comment by squidy669** [comment]
was not expecting an interesting read, i should imagine that MS may have fairs and would like to keep consumers on they're platform.. i was fairly taken aback that i had to enter a code for unlocking secure boot and considering most if not all systems will now be using secure boot that at its core is good but when it comes to dual booting or tinkering will become a pain for any windows 11 users its started me thinking that there is a bigger plan.

**Reply by StollD** [reply]
> i had to enter a code for unlocking secure boot

Thats not for secure boot, but for bitlocker. Bitlocker encrypts the harddrive with a key that is bound to the state of your system, this includes secure boot. If you disable secure boot, that key stops working, so you need to decrypt your disk differently.

If you wipe windows, disable bitlocker before changing secure boot, or keep secure boot active (most major distros support it), then you wont need a code.

**Reply by squidy669** [reply]
and there was me thinking windows were enforcing anti consumer/pro monopoly  practices.
